lib/game_client/ogs/ogs_game_client.dart

```diff
@@ -153,15 +252,114 @@ class OGSGameClient extends GameClient {
 
   @override
   Future<Game> findGame(String presetId) async {
-    throw UnimplementedError();
+    // Parse the preset ID to extract board size and speed (format: "9_blitz", "19_live", etc.)
+    final parts = presetId.split('_');
+    final boardSize = int.parse(parts[0]);
+    final speed = parts.length >= 2 ? parts[1] : 'rapid';
+
+    // Generate a UUID for the automatch request and store it for later cancellation
+    _currentAutomatchUuid = const Uuid().v4();
+    _automatchCompleter = Completer<Game>();
+
+    // Set up message listener for automatch responses
+    _messageSubscription =
+        _webSocketManager.messages.listen(_handleAutomatchResponse);
+
+    // Create the automatch payload data matching OGS format
+    final automatchData = {
+      "uuid": _currentAutomatchUuid!,
+      "size_speed_options": [
+        {"size": "${boardSize}x$boardSize", "speed": speed, "system": "byoyomi"}
+      ],
+      "lower_rank_diff": 3,
+      "upper_rank_diff": 3,
+      "rules": {"condition": "required", "value": "japanese"},
+      "handicap": {"condition": "preferred", "value": "enabled"}
+    };
+
+    // Send the automatch request via WebSocket
+    _webSocketManager.send('automatch/find_match', automatchData);
+
+    return _automatchCompleter!.future;
+  }
+
+  void _handleAutomatchResponse(Map<String, dynamic> message) async {
+    // Check if this is an automatch/start message
+    if (message['event'] == 'automatch/start' &&
+        message['data']['uuid'] == _currentAutomatchUuid) {
+      final gameId = message['data']['game_id'] as int;
+
+      // Clean up resources
+      _messageSubscription?.cancel();
+      _messageSubscription = null;
+      _currentAutomatchUuid = null;
+
+      final response = await _httpClient.get(
+        Uri.parse('$serverUrl/api/v1/games/$gameId'),
+        headers: {
+          'User-Agent': _userAgent,
+          if (_csrfToken != null) 'X-CSRFToken': _csrfToken!,
+        },
+      );
+
+      Map<String, dynamic> gameData = {};
+      if (response.statusCode == 200) {
+        gameData = jsonDecode(response.body)['gamedata'];
```

Comment:
We may be able to avoid the http request for gamedata here, since WS will send it to us once we've connected to the game.  However, `Game` constructor requires values such as `size`, `handicap`, `komi` so that's why I do the fetch here.

Reply:
yeah I guess the other option would be to wait for that first message before creating the game. But if it's fast enough with the extra request, no big deal, as long as it's reliable.